### PR TITLE
Doc: Update the developer guide for the v3

### PR DESCRIPTION
**Pull request by anyangml**

### CODE REVIEW

File: doc/development/create-a-model.md [hunk 1]
```diff
@@ -22,7 +25,119 @@ class SomeDescript(Descriptor):
     def __init__(self, arg1: bool, arg2: float) -> None:
         pass
 ```
+:::
+
+:::{tab-item} PyTorch {{ pytorch_icon }}
+The PyTorch backend follows a meticulously structured inheritance pattern. When creating a new descriptor, it is essential to inherit from both the {py:class}`deepmd.pt.model.descriptor.base_descriptor.BaseDescriptor` class and the {py:class}`torch.nn.Module` class. Abstract methods, including {py:class}`deepmd.pt.model.descriptor.base_descriptor.BaseDescriptor.fwd`, must be implemented, while others remain optional. It is crucial to adhere to the original method arguments without any modifications. Once the implementation is complete, the next step involves registering the component with a designated key:
```
Comment (wanghan-iapcm) [inline doc/development/create-a-model.md:31]:
```suggestion
The PyTorch backend follows a meticulously structured inheritance pattern. When creating a new descriptor, it is essential to inherit from both the {py:class}`deepmd.pt.model.descriptor.base_descriptor.BaseDescriptor` class and the {py:class}`torch.nn.Module` class. Abstract methods, including {py:class}`deepmd.pt.model.descriptor.base_descriptor.BaseDescriptor.forward`, must be implemented, while others remain optional. It is crucial to adhere to the original method arguments without any modifications. Once the implementation is complete, the next step involves registering the component with a designated key:
```

Comment (wanghan-iapcm) [reply doc/development/create-a-model.md:31]: please check https://github.com/deepmodeling/deepmd-kit/blob/a7d44d1c2283bcc4a096b1f0eb5eaf6241078ef1/deepmd/pt/model/descriptor/base_descriptor.py#L8 that rename the name of the forward method.

Comment (wanghan-iapcm) [reply doc/development/create-a-model.md:31]: many abstract methods needs to be implemented. please check deepmd/dpmodel/descriptor/make_base_descriptor.py

File: doc/development/create-a-model.md [hunk 2]
```diff
@@ -22,7 +25,119 @@ class SomeDescript(Descriptor):
...
+    ):
+        pass
+
+    def output_def(self) -> FittingOutputDef:
+        pass
+```
+
+The model architecture within the PyTorch backend is structured with multiple layers of abstraction to provide a high degree of flexibility. Generally, the process begins with an atomic model responsible for handling atom-wise property calculations. This atomic model should inherit from the {py:class}`deepmd.pt.model.atomic_model.base_atomic_model.BaseAtomicModel` class and the {py:class}`torch.nn.Module` class.
+
+Subsequently, the `AtomicModel` is encapsulated using the `make_model(AtomicModel)` function, employing the `deepmd.pt.model.model.make_model.make_model` function. The purpose of the `make_model` wrapper is to facilitate the translation between the original system and the extended system.
```
Comment (wanghan-iapcm) [inline doc/development/create-a-model.md:101]:
```suggestion
Subsequently, the `AtomicModel` is encapsulated using the `make_model(AtomicModel)` function, employing the `deepmd.pt.model.model.make_model.make_model` function. The purpose of the `make_model` wrapper is to facilitate the translation between the atomic  property prediction and the extended property prediction and differentiation , e.g. the reduction of atomic energy contribution and the autodiff for calculating the forces and virial. The developers usually need to implement an `AtomicModel` not a `Model`.
```

File: doc/development/create-a-model.md [hunk 3]
```diff
@@ -22,7 +25,119 @@ class SomeDescript(Descriptor):
...
+
+Finally, the entire model is wrapped within a `DPModel`, which must inherit from the {py:class}`deepmd.pt.model.model.model.BaseModel` class and include the aforementioned `make_model(AtomicModel)`. The user directly interacts with a wrapper built on top of the `DPModel` to seamlessly handle result translations.
```
Comment (wanghan-iapcm) [inline doc/development/create-a-model.md:103]: Not necessary? The DPZBL model is not inherited from the `DPModel`.

Comment (wanghan-iapcm) [reply doc/development/create-a-model.md:103]: the `DPModel` stands for the models that have the descriptor-fitting architecture.

File: doc/development/create-a-model.md [hunk 4]
```diff
@@ -22,7 +25,119 @@ class SomeDescript(Descriptor):
...
+    def forward_atomic(self):
+        pass
+
+
+@BaseModel.register("some_model")
+class SomeDPModel(make_model(SomeAtomicModel), BaseModel):
+    pass
+
+
+class SomeModel(SomeDPModel):
```
Comment (wanghan-iapcm) [inline doc/development/create-a-model.md:130]: The out-most model is only for key translation of the returned dict.

**Review by wanghan-iapcm — COMMENTED** [review]
- The way of creating a new model in dp/pt looks very different from the old tf manner. shall we provide a separate doc for dp/pt model development? 
- we may introduce the idea of how to maintain models across platforms, then introduce the dp backend.
- please tell the developers how to write UTs for cross-platform consistency.

File: doc/development/create-a-model.md (hunk 3 above)
Comment (anyangml) [reply doc/development/create-a-model.md:103]: DPZBL has it's own `DPModel` called DPZBLModel_? maybe I should use a different keyword.

File: doc/development/create-a-model-pt.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+```py
+entry_points = (
+    {
+        "deepmd": [
+            "some_descrpt=deepmd_some_descrtpt:SomeDescript",
+        ],
+    },
+)
+```
```
[the remark this answers is not in the document]
Comment (njzjz) [reply doc/development/create-a-model-pt.md:187]: Btw, I am not sure whether this article aims to contribute to the deepmd-kit repo or develop a plugin outside the repo.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: doc/development/create-a-model-pt.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (anyangml) [reply doc/development/create-a-model-pt.md:187]: we remove this section then?

**Review by wanghan-iapcm — APPROVED** [review]

File: doc/development/create-a-model-pt.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    ):
+        pass
+
+    def output_def(self) -> FittingOutputDef:
+        pass
+```
+### New models
+The PyTorch backend's model architecture is meticulously structured with multiple layers of abstraction, ensuring a high degree of flexibility. Typically, the process commences with an atomic model responsible for atom-wise property calculations. This atomic model inherits from both the {py:class}`deepmd.pt.model.atomic_model.base_atomic_model.BaseAtomicModel` class and the {py:class}`torch.nn.Module` class.
+
+Subsequently, the `AtomicModel` is encapsulated using the `make_model(AtomicModel)` function, which leverages the `deepmd.pt.model.model.make_model.make_model` function.  The purpose of the `make_model` wrapper is to facilitate the translation between atomic property predictions and the extended property predictions and differentiation , e.g. the reduction of atomic energy contribution and the autodiff for calculating the forces and virial. The developers usually need to implement an `AtomicModel` not a `Model`.
```
Comment (iProzd) [inline doc/development/create-a-model-pt.md:110]: shall we add the inheritance of forward functions in different level? e.g. forward --> forward_common --> forward_common_lower --> forward_common_atomic --> forward_atomic

**Review by iProzd — APPROVED** [review]

**Review by njzjz — APPROVED** [review]

File: doc/development/create-a-model-pt.md (hunk 6 above)
Comment (anyangml) [reply doc/development/create-a-model-pt.md:110]: As discussed, in most cases the developer only need to handle AtomicModel, thus only cares about `forward_atomic`.

**Pull request merged**